Add release notes for 0.7.3 (#910)
scalameta/metals: website/blog/2019-09-11-thorium.md

In this release we merged 10 PRs and closed 6 issues, full details:
https://github.com/scalameta/metals/milestone/15?closed=1

## TL;DR

Check out the website and give Metals a try: https://scalameta.org/metals/

- add auto-fill option to case classes
- deduce values for named parameters completions
- multiple fixes after adding support for Scala 2.12.9
- fixes for automatic addition of `|` in multiline strings

## Add auto-fill option to case classes

Whenever we invoke a method and try completions inside the brackets there is an
additional option to allow Metals to deduce what should be filled for each
parameter.

![auto-fill](https://i.imgur.com/8pCiMqE.gif)

This algorithm checks if there exists a value in scope with the same type as the
parameters' and then works according to the rules below:

```
1. If there is only one candidate, use it.
2. If there is more than one, give user an alternative to choose one of them.
3. Otherwise use use "???".
```

This feature also uses the snippets syntax to allow to quickly go over the
auto-filled parameters and choose or replace the auto-filled values.

## Deduce values for named parameters completions

> As a side effect of implementing the previous feature, we were able to provide

Reviewer: ```suggestion
Leveraging the work that went into the previous feature, we were also able to provide
```

additional support for concrete values in named parameter completions.

![deduce](https://i.imgur.com/m11xixy.gif)

We generate completion item for each value that fits the type of the parameter.

## Multiple fixes after adding support for Scala 2.12.9

> After updating Metals to work with Scala 2.12.9 it turned out that completions

Reviewer: ```suggestion
After updating Metals to work with Scala 2.12.9 it turned out that completions and hover
```

> didn't work correctly and crached the presentation compiler in the following

Reviewer: ```suggestion
didn't work correctly, due to the presentation compiler crashing in the following
```
